src/coreclr/jit/flowgraph.cpp

```diff
@@ -1388,14 +1388,9 @@ void Compiler::fgAddSyncMethodEnterExit()
     NYI("No support for synchronized methods");
 #endif // !FEATURE_EH
 
-    // Create a scratch first BB where we can put the new variable initialization.
-    // Don't put the scratch BB in the protected region.
-
-    fgEnsureFirstBBisScratch();
-
     // Create a block for the start of the try region, where the monitor enter call
     // will go.
-    BasicBlock* const tryBegBB  = fgSplitBlockAtEnd(fgFirstBB);
+    BasicBlock* const tryBegBB  = fgSplitBlockAtBeginning(fgFirstBB);
```

Comment:
Yeah, I agree it looks a bit odd, but I think it's preferable over a solution that tries to differentiate between IR created because of the user IL and IR created due to the JIT. I really want us to be able to freely move IR and merge/duplicate blocks without breaking subtle invariants about what kind of IR can go where.

So far it seems like most of these cases come down to careful consideration of phase ordering. For example, this try-finally insertion is very much like IR created because of IL, so it would make sense for me to do it in (or close to) import. Another example is reverse pinvoke enter/exit – most of the JIT will naturally assume inserted IR is executed in cooperative mode, so this one makes sense to insert as the last thing in the JIT. Currently it's inserted quite early.